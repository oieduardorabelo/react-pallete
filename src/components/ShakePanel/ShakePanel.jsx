import React from 'react';

import { stopPropagation } from '../../domain';

import styles from '../ColorColumn/ColorColumnStyles'

const panelStyles = styles.panel();
const panelOverlayStyles = styles.panelOverlay();

const displayFlex = 'flex';
panelStyles.display = displayFlex;
panelOverlayStyles.display = displayFlex;

const ShakePanel = (props) => (
  <div>
    <div style={panelOverlayStyles} onClick={props.onClickToggleShakePanel} />

    <div style={panelStyles} onClick={stopPropagation}>
      <button
        onClick={props.onClickToggleShakePanel}
        style={styles.panelClose()}
        type="button"
      >
        Close
      </button>

      <span>Shake it off!</span>
      <p>When you shake your phone, we set random colors to your pallete!</p>
    </div>
  </div>
)

ShakePanel.propTypes = {
  onClickToggleShakePanel: React.PropTypes.func.isRequired,
}

export default ShakePanel;
