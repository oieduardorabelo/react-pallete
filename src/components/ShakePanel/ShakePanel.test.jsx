import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ShakePanel from './ShakePanel';

it('renders correctly', () => {
  const wrapper = mount(
    <ShakePanel onClickToggleShakePanel={jest.fn()} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

it('calls correct onClick spy', () => {
  const spy = jest.fn();
  const wrapper = mount(
    <ShakePanel onClickToggleShakePanel={spy} />
  );

  wrapper.find('button').simulate('click');

  expect(spy).toHaveBeenCalledTimes(1);
});
