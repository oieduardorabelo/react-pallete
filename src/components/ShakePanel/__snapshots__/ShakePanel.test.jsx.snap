exports[`test renders correctly 1`] = `
<ShakePanel
  onClickToggleShakePanel={[Function]}>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "animation": "fadeIn 160ms linear",
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "bottom": 0,
          "display": "flex",
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      } />
    <div
      onClick={[Function]}
      style={
        Object {
          "animation": "slideIn 100ms linear",
          "backgroundColor": "#FFFFFF",
          "bottom": 0,
          "display": "flex",
          "flexDirection": "column",
          "left": 0,
          "padding": 20,
          "position": "fixed",
          "right": 0,
          "zIndex": 2,
        }
      }>
      <button
        onClick={[Function]}
        style={
          Object {
            "alignSelf": "flex-end",
          }
        }
        type="button">
        Close
      </button>
      <span>
        Shake it off!
      </span>
      <p>
        When you shake your phone, we set random colors to your pallete!
      </p>
    </div>
  </div>
</ShakePanel>
`;
